sklearn/utils/estimator_checks.py

```diff
@@ -164,6 +165,10 @@ def _yield_classifier_checks(classifier):
         yield check_estimators_unfitted
     if "class_weight" in classifier.get_params().keys():
         yield check_class_weight_classifiers
+        if has_fit_parameter(classifier, "sample_weight") and not tags["pairwise"]:
```

Comment:
This is not obvious why the test operates on `tags["pairwise"]`.

Can you add a comment precising it?

Reply:
I added it on aab209a trying to solve some xfail logic, but I removed it on this last commit and I did not get any error. I think it is not needed (?).